yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -1390,24 +1390,49 @@ func IsCurrentUserSuperUser(tconf *TargetConf) (bool, error) {
 	}
 	defer conn.Close(context.Background())
 
-	query := "SELECT rolsuper FROM pg_roles WHERE rolname=current_user"
-	rows, err := conn.Query(context.Background(), query)
-	if err != nil {
-		return false, fmt.Errorf("querying if user is superuser: %w", err)
-	}
-	defer rows.Close()
-
-	var isSuperUser bool
-	if rows.Next() {
-		err = rows.Scan(&isSuperUser)
+	runQueryAndCheckPrivilege := func(query string) (bool, error) {
+		rows, err := conn.Query(context.Background(), query)
 		if err != nil {
-			return false, fmt.Errorf("scanning row for superuser: %w", err)
+			return false, fmt.Errorf("querying if user is superuser: %w", err)
 		}
-	} else {
-		return false, fmt.Errorf("no current user found in pg_roles")
+		defer rows.Close()
+
+		var isProperUser bool
+		if rows.Next() {
+			err = rows.Scan(&isProperUser)
+			if err != nil {
+				return false, fmt.Errorf("scanning row for query: %w", err)
+			}
+		} else {
+			return false, fmt.Errorf("no current user found in pg_roles")
+		}
+		return isProperUser, nil
 	}
 
-	return isSuperUser, nil
+	isSuperUserquery := "SELECT rolsuper FROM pg_roles WHERE rolname=current_user"
+
+	isSuperUser, err := runQueryAndCheckPrivilege(isSuperUserquery)
+
+	if isSuperUser {
+		return true, nil
+	}
+	//In case of YugabyteDB Aeon deployment of target database we need to verify if yb_superuser is granted or not
+	isYbSuperUserQuery := `SELECT 
+    CASE 
+        WHEN EXISTS (
+            SELECT 1
+            FROM pg_auth_members m
+            JOIN pg_roles grantee ON m.member = grantee.oid
+            JOIN pg_roles granted ON m.roleid = granted.oid
+            WHERE grantee.rolname = CURRENT_USER AND granted.rolname = 'yb_superuser'
+        ) 
+        THEN TRUE 
+        ELSE FALSE 
+    END AS is_yb_superuser;`
+
+	isYBSuperUser, err := runQueryAndCheckPrivilege(isYbSuperUserQuery)
```

Comment:
Also can we ensure that this query will return false if the `yb_superuser` property is not there in a db version and not error out?

Reply:
yes, in case of yb_superuser role is not present in the pg_roles it returns true 
```
yugabyte=# SELECT 
    CASE 
        WHEN EXISTS (
            SELECT 1
            FROM pg_auth_members m
            JOIN pg_roles grantee ON m.member = grantee.oid
            JOIN pg_roles granted ON m.roleid = granted.oid
            WHERE grantee.rolname = CURRENT_USER AND granted.rolname = 'yb_superuser'
        ) 
        THEN TRUE 
        ELSE FALSE 
    END AS is_yb_superuser;
 is_yb_superuser 
-----------------
 f
(1 row)

yugabyte=# select * from pg_roles where rolname='yb_superuser';
 rolname | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolconnlimit | rolpassword | rolvaliduntil | rolbypassrls | rolconfig | oid 
---------+----------+------------+---------------+-------------+-------------+----------------+--------------+-------------+---------------+--------------+-----------+-----
(0 rows)

callhome=# 
```